yew-router/examples/minimal/src/main.rs

```diff
@@ -36,9 +36,9 @@ pub enum AppRoute {
 }
 
 impl Model {
-    fn change_route(&self, app_route: AppRoute) -> Callback<MouseEvent> {
+    fn change_route(&self, app_route: AppRoute) -> Callback<ClickEvent> {
         self.link.callback(move |_| {
-            let route = app_route.clone(); // TODO, I don't think I should have to clone here?
```

Comment:
FYI: We now can create FnOnce callbacks where you wouldn't need to clone here. But in this example it looks like that wouldn't be advised because someone could click the button fast and trigger two callback calls before render

Reply:
Neat, what would happen if someone performs that double-click? If it's a no-op, then I think that would be acceptable in this situation.

Reply:
It's a panic

Reply:
In that case, I'll leave it as a clone here.

Without knowing the details of how it is currently implemented - so I don't know if its possible, I think there is a reasonable use case for a callback variant that can only be activated once-per render, and then just no-ops, because ignoring a second click in a situation like this would actually be a preferred behavior.

Reply:
Cool, seems reasonable to me! Thanks for the feedback